packages/blade/src/components/Tag/_decisions/decisions.md

```diff
@@ -0,0 +1,198 @@
+
+```ts
+type TagProps = {
+  /**
+   * Decides the size of Tag
+   *
+   * @default medium
+   */
+  size?: 'medium' | 'large';
+
+  /**
+   * Callback when close icon on Tag is clicked
+   *
+   * Gives a `value` argument which has same text as `children`
+   */
+  onDismiss?: ({ value }: { value: TagProps['children'] }) => void;
+
+  /**
+   * Text that renders inside Tag
+   */
+  children: ReactText;
+
+  /**
+   * Disable tag
+   */
+  isDisabled?: boolean;
+};
+```
+
+_Didn't consider any alternate API. Kept `size` and `children` similar to `<Badge />` and `onDismiss` similar to `<Alert />` and `<Dropdown />`_
+
+## Integration with Inputs
+
+<img width="358" alt="image" src="https://github.com/razorpay/blade/assets/30949385/71cd236a-93e7-46f0-9da6-12d519b958d4">
+
+Tags can be added inside other input components such as -
+
+- TextArea
+- SelectInput (in Dropdown)
+- TextInput
+- AutoComplete
+
+<details>
+<summary>Implementation Details</summary>
+
+This will require some refactor in BaseInput to add a slot before the actual Input element.
+
+E.g. This is AutoComplete from Primer where the tags go into a slot and input gets pushed forward. We will be implementing something similar in our BaseInput.
+
+<img width="419" alt="image" src="https://github.com/razorpay/blade/assets/30949385/be01cdfb-1171-4970-b381-76e4c44d1854">
+
+</details>
+
+### Integration with SelectInput and AutoComplete
+
+SelectInput and AutoComplete are not expected to have any changes in consumer API as rendering of tags will be handled internally based on the selected value.
+
+### Integration with TextInput and TextArea
+
+> **Note**
+>
+> This implementation is out of scope for initial release of Tag.
+>
+> Have added considered approaches below which we can refer back and confirm the feasibility while implementation. (Feel free to review and suggest any changes here though. We can consider them while imeplementation in future)
+
+On consumer end, the APIs would look like -
+
+#### New `tagsSlot` prop on `TextInput` and `TextArea`
+
+- Can only be used in Controlled Input
+  - Why?
+    - This is comparitively a rare usecase so don't see any need of having another `hasTags` props that handles adding tags internally.
+
+```jsx
+<TextInput tagsSlot={} value="" onChange={} />
+```
+
+##### Alternate Prop Name
+
+- `leading` (felt confusing because it is leading to the invisible input tag and not the actual visible TextInput)
+- `valuePrefix` (Similar to `titleSuffix` from BaseHeader)
+
+<details>
+<summary>Full Controlled Example</summary>
+
+```jsx
+function App() {
+  const [inputValue, setInputValue] = React.useState('');
+  const [tags, setTags] = React.useState([]);
+
+  const addTag = () => {
+    // Add input value to tags and clear the input value
+    setTags([...tags, inputValue]);
+    setInputValue('');
+  }
+
+  const removeTag = (tagName) => {
+    setTags(tags.filter(tagNameValue) => tagNameValue !== tagName);
+  }
+
+  return (
+    <TextInput
```

Comment:
How about adding tagsSlot on BaseTextInput only and using that for our internal components? This way we can avoid exposing this to consumers

Reply:
We might have to expose to our consumers - https://github.com/razorpay/blade/pull/1315#discussion_r1247724657